### PR TITLE
refactor MessageSenderLib

### PR DESCRIPTION
**Pull request by patrickmao93**

### CODE REVIEW

File: contracts/message/libraries/MessageSenderLib.sol [hunk 1]
```diff
@@ -68,208 +63,143 @@ library MessageSenderLib {
...
+            IERC20(_token).safeIncreaseAllowance(bridge, _amount);
+            IPeggedTokenBridge(bridge).burn(_token, _amount, _receiver, _nonce);
+            // handle cases where certain tokens do not spend allowance for role-based burn
+            IERC20(_token).safeApprove(bridge, 0);
+            transferId = computePegV1BurnId(_receiver, _token, _amount, _nonce);
+        } else if (_bridgeSendType == MsgDataTypes.BridgeSendType.PegV2Deposit) {
+            bridge = IMessageBus(_messageBus).pegVaultV2();
+            IERC20(_token).safeIncreaseAllowance(bridge, _amount);
+            IOriginalTokenVaultV2(bridge).deposit(_token, _amount, _dstChainId, _receiver, _nonce);
+            transferId = computePegV2Id(_receiver, _token, _amount, _dstChainId, _nonce);
```
Comment (xiaozhou) [inline contracts/message/libraries/MessageSenderLib.sol:129]: use returned Id from the v2 contract

File: contracts/message/libraries/MessageSenderLib.sol [hunk 2]
```diff
@@ -68,208 +63,143 @@ library MessageSenderLib {
...
+            keccak256(
+                abi.encodePacked(
+                    msg.sender,
+                    _token,
+                    _amount,
+                    _dstChainId,
+                    _receiver,
+                    _nonce,
+                    uint64(block.chainid),
+                    address(this)
```
Comment (xiaozhou) [inline contracts/message/libraries/MessageSenderLib.sol:201]: This is incorrect, the last address field is the bridge/vault contract address. Anyway, remove this function, use returned Id from the v2 contract when needed.

File: contracts/message/libraries/MessageSenderLib.sol [hunk 3]
```diff
@@ -68,208 +63,143 @@ library MessageSenderLib {
...
+            _receiver,
+            _token,
+            _amount,
+            _dstChainId,
+            _nonce,
+            _maxSlippage,
+            _bridgeSendType,
+            _messageBus
+        );
+        IMessageBus(_messageBus).sendMessageWithTransfer{value: _fee}(
```
Comment (xiaozhou) [inline contracts/message/libraries/MessageSenderLib.sol:76]: current lib only send message `if (_message.length > 0)`

File: contracts/message/libraries/MessageSenderLib.sol [hunk 4]
```diff
@@ -68,208 +63,143 @@ library MessageSenderLib {
...
-     *        If message is empty, only the token transfer will be sent
-     * @param _messageBus The address of the MessageBus on this chain.
-     * @param _fee The fee amount to pay to MessageBus.
-     * @return The transfer ID.
+     * Must be greater than minimalMaxSlippage. Receiver is guaranteed to receive at least (100% - max slippage percentage) * amount or the
+     * transfer can be refunded.
+     * @param _bridgeSendType One of the {MsgDataTypes.BridgeSendType} enum.
      */
-    function sendMessageWithLiquidityBridgeTransfer(
+    function sendTokenTransfer(
```
Comment (xiaozhou) [inline contracts/message/libraries/MessageSenderLib.sol:98]: looks similar to https://github.com/celer-network/sgn-v2-contracts/blob/main/contracts/libraries/BridgeTransferLib.sol#L64

Comment (patrickmao93) [reply contracts/message/libraries/MessageSenderLib.sol:98]: I only extracted the "token transfer" part from sendMessageWithXXX functions. We could use BridgeTransferLib in MessageSenderLib, but doing this requires the user of this lib to also copy 3 XXX.Pb.sol files, even if they don't use those

File: contracts/message/libraries/MessageSenderLib.sol (hunk 2 above)
Comment (patrickmao93) [reply contracts/message/libraries/MessageSenderLib.sol]: fixed

File: contracts/message/libraries/MessageSenderLib.sol (hunk 1 above)
Comment (patrickmao93) [reply contracts/message/libraries/MessageSenderLib.sol]: fixed

File: contracts/message/libraries/MessageSenderLib.sol (hunk 3 above)
Comment (patrickmao93) [reply contracts/message/libraries/MessageSenderLib.sol]: fixed

**Review by xiaozhou — APPROVED** [review]

**Pull request merged**